### ISSUE TITLE
Pager not made for list_collection_ids proto on firestore, seems heuristic may be incorrect?

### ISSUE THREAD

**Issue by crwilcox** [googleapis/gapic-generator-python#516]
```
def collections(self):
        """List top-level collections of the client's database.

        Returns:
            Sequence[:class:`~google.cloud.firestore_v1.collection.CollectionReference`]:
                iterator of subcollections of the current document.
        """
        iterator = self._firestore_api.list_collection_ids(
            request={"parent": "{}/documents".format(self._database_string)},
            metadata=self._rpc_metadata,
        )
        iterator.client = self
        iterator.item_to_value = _item_to_collection_ref
        return iterator
```

This code is how we used ot use the list_collection_ids. However, it is no longer a pager/iterator.
 
https://github.com/googleapis/googleapis/blob/master/google/firestore/v1/firestore.proto#L842

**Comment by software-dov** [comment]
It looks like the problem is that the generator is expecting that APIs will only paginate over messages, not primitives. The repeated element for `ListCollectionIdsResponse` is a collection of strings, and so the heuristic ignores it.
I don't know if this was an oversight or a deliberate decision when implementing the generator logic. I'll check to see if there are any tests that would support one hypothesis or another.

**Comment by software-dov** [comment]
FIxed by #517
